src/views/MissionPlanningView.vue

```diff
@@ -753,6 +780,14 @@ const addWaypointConnection = (fromWaypoint: Waypoint, toWaypoint: Waypoint): vo
   planningMap.value?.addLayer(polyline)
 }
 
+onMounted(async () => {
+  await nextTick()
+  if (planningMap.value) {
+    planningMap.value.on('contextmenu', showContextMenu)
+    planningMap.value.on('click', hideContextMenu)
```

Comment:
Is this click listener working fine in parallel with the other listener on line 792?

Seems like a better approach would be to centralize the listener on a "onMapClick" function and deal with the possible cases in this function. This way it is explicit what is going to happen on each situation (e.g.: close context-menu, add new waypoint, etc).